builtin/logical/transit/path_decrypt.go

```diff
@@ -84,24 +149,42 @@ func (b *backend) pathDecryptWrite(
 		return logical.ErrorResponse("policy not found"), logical.ErrInvalidRequest
 	}
 
-	plaintext, err := p.Decrypt(context, nonce, ciphertext)
-	if err != nil {
-		switch err.(type) {
-		case errutil.UserError:
-			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
-		case errutil.InternalError:
-			return nil, err
-		default:
-			return nil, err
+	for i, item := range batchInputItems {
+		if batchResponseItems[i].Error != "" {
+			continue
+		}
+
+		plaintext, err := p.Decrypt(item.Context, item.Nonce, item.Ciphertext)
+		if err != nil {
+			switch err.(type) {
+			case errutil.UserError:
+				batchResponseItems[i].Error = err.Error()
+				continue
+			default:
+				return nil, err
+			}
 		}
+		batchResponseItems[i].Plaintext = plaintext
 	}
 
-	// Generate the response
-	resp := &logical.Response{
-		Data: map[string]interface{}{
-			"plaintext": plaintext,
-		},
+	resp := &logical.Response{}
+	if len(batchInputRaw) != 0 {
+		batchResponseJSON, err := jsonutil.EncodeJSON(batchResponseItems)
+		if err != nil {
+			return nil, fmt.Errorf("failed to JSON encode batch response")
+		}
+		resp.Data = map[string]interface{}{
+			"data": string(batchResponseJSON),
```

Comment:
Maybe name this `batch_results` to correspond to `batch` on the input side? Maybe it'd be nice to have `batch` on the input side actually be `batch_input` but I don't feel strongly about it.